include/boost/geometry/views/detail/boundary_view/implementation.hpp

```diff
@@ -0,0 +1,400 @@
+    inline Reference dereference() const
+    {
+        if (m_index == 0)
+        {
+            return exterior_ring(*m_polygon);
+        }
+        return range::at(interior_rings(*m_polygon), m_index - 1);
+    }
+
+    template
+    <
+        typename OtherPolygon,
+        typename OtherValue,
+        typename OtherReference,
+        typename OtherDifference
+    >
+    inline bool equal(polygon_rings_iterator
+                      <
+                          OtherPolygon,
+                          OtherValue,
+                          OtherReference,
+                          OtherDifference
+                      > const& other) const
+    {
+        return m_polygon == other.m_polygon && m_index == other.m_index;
+    }
+
+    inline void increment()
+    {
+        ++m_index;
+    }
+
+    inline void decrement()
+    {
+        --m_index;
+    }
+
+    template
+    <
+        typename OtherPolygon,
+        typename OtherValue,
+        typename OtherReference,
+        typename OtherDifference
+    >
+    inline Difference distance_to(polygon_rings_iterator
+                                  <
+                                      OtherPolygon,
+                                      OtherValue,
+                                      OtherReference,
+                                      OtherDifference
+                                  > const& other) const
+    {
+        return other.m_index - m_index;
+    }
+
+    inline void advance(Difference n)
+    {
+        m_index += n;
+    }
+
+private:
+    Polygon* m_polygon;
+    std::size_t m_index;
+};
+
+
+template <typename Ring>
+class ring_boundary : closeable_view<Ring, closure<Ring>::value>::type
+{
+private:
+    typedef typename closeable_view<Ring, closure<Ring>::value>::type base_type;
+
+public:
+    typedef typename base_type::iterator iterator;
+    typedef typename base_type::const_iterator const_iterator;
+    
+    typedef linestring_tag tag_type;
+
+    explicit ring_boundary(Ring& ring)
+        : base_type(ring) {}
+
+    iterator begin() { return base_type::begin(); }
+    iterator end() { return base_type::end(); }
+    const_iterator begin() const { return base_type::begin(); }
+    const_iterator end() const { return base_type::end(); }
+};
+
+
+template <typename Polygon>
+class polygon_boundary
+{
+    typedef boundary_view<typename ring_type<Polygon>::type> boundary_view_type;
+    typedef polygon_rings_iterator<Polygon> rings_iterator_type;
+
+    inline void initialize_views(Polygon& polygon)
+    {
+        std::pair<boundary_view_type*, std::ptrdiff_t> result
+            = std::get_temporary_buffer<boundary_view_type>(m_num_rings);
+
+        if ( result.second == 0 )
+        {
+            throw std::bad_alloc();
+        }
+
+        m_views = result.first;
+        std::uninitialized_copy(rings_iterator_type(polygon),
+                                rings_iterator_type(polygon, true),
+                                m_views);
+    }
```

Comment:
Great! Still this is not yet exception safe. If one of the constructors called in `uninitialized_copy` thrown an exception this exception would be propagated into the `polygon_boundary` ctor and the object would not be considered as created. The destructor would not be called in this case. Which means that the memory allocated with `get_temporary_buffer` would not be released. You could use RAII and have an automatic destroyer (similar to smart ptr), automatically deallocating the memory on destruction inside this function (if the exception was thrown this object would be destroyed). Or you could just put a try-catch block there and rethrow an exception. And since this pattern is also repeated in the implementation for MultiPolygon, you could implement a simple class for this scoped storage.